api/Services/Files/CivilFilesService.cs

```diff
@@ -358,6 +355,51 @@ public async Task<CivilFileContent> FileContentAsync(string agencyId, string roo
                 appearanceId, physicalFileId);
         }
 
+        public async Task<IList<CivilDocument>> GetBinderDocumentsAsync(string fileId)
+        {
+            async Task<CivilFileDetailResponse> FileDetails() => await _filesClient.FilesCivilGetAsync(_requestAgencyIdentifierId, _requestPartId, _applicationCode, fileId);
+            async Task<CivilFileContent> FileContent() => await _filesClient.FilesCivilFilecontentAsync(_requestAgencyIdentifierId, _requestPartId, _applicationCode, null, null, null, null, fileId);
+
+            var fileDetailTask = _cache.GetOrAddAsync($"GetBinderDocumentsAsync-CivilFileDetail-{fileId}-{_requestAgencyIdentifierId}", FileDetails);
+            var fileContentTask = _cache.GetOrAddAsync($"GetBinderDocumentsAsync-CivilFileContent-{fileId}-{_requestAgencyIdentifierId}", FileContent);
+
+            await Task.WhenAll(fileDetailTask, fileContentTask);
+
+            var detail = _mapper.Map<RedactedCivilFileDetailResponse>(fileDetailTask.Result);
+            var fileContentCivilFile = fileContentTask.Result?.CivilFile?.First(cf => cf.PhysicalFileID == fileId);
+
+            var labels = new Dictionary<string, string>
+            {
+                { LabelConstants.PHYSICAL_FILE_ID, detail.PhysicalFileId },
+                { LabelConstants.COURT_CLASS_CD, detail.CourtClassCd.ToString() },
+                { LabelConstants.JUDGE_ID, _currentUser.UserId().ToString() }
+            };
+
+            var binders = await _binderService.GetByLabels(labels);
+
+            if (!binders.Succeeded || binders.Payload.Count == 0)
+            {
+                return [];
+            }
+
+            var binder = binders.Payload[0];
+            var binderDocIds = binder.Documents.Select(d => d.DocumentId);
+
+            var binderDocIdsOrdered = binder.Documents.Select((d, index) => new { d.DocumentId, Order = index }).ToDictionary(x => x.DocumentId, x => x.Order);
+
+            var csrDocs = PopulateDetailCsrsDocuments([.. fileDetailTask.Result.Appearance.Where(a => binderDocIds.Contains(a.AppearanceId))]);
+
+            var mappedDetail = _mapper.Map<RedactedCivilFileDetailResponse>(detail);
```

Comment:
detail comes from this mapping:

var detail = _mapper.Map<RedactedCivilFileDetailResponse>(fileDetailTask.Result);

mapping it again doesn't feel correct.